client/src/components/admin/GeneralNotification/GeneralNotificationList.tsx

```diff
@@ -0,0 +1,161 @@
+  },
+});
+
+interface Props {
+  onEdit: (notification: GeneralNotification) => void;
+  onCancel: () => void;
+  activeNotification: GeneralNotification | null;
+  notifications: GeneralNotification[];
+  editing: boolean;
+  loading: boolean;
+  editingEnabled: boolean;
+}
+
+export function GeneralNotificationList({
+  onEdit,
+  onCancel,
+  notifications,
+  activeNotification,
+  loading,
+  editing,
+  editingEnabled,
+}: Props) {
+  const { tr } = useTranslations();
+  const theme = useTheme();
+
+  if (loading) {
+    return (
+      <CircularProgress
+        sx={{
+          height: '150px',
+          margin: 'auto',
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'center',
+          alignItems: 'center',
+        }}
+      />
+    );
+  }
+
+  if (notifications.length === 0) {
+    return (
+      <Box
+        sx={{
+          height: '150px',
+          margin: 'auto',
+          display: 'flex',
+          gap: '0.5rem',
+          flexDirection: 'column',
+          justifyContent: 'center',
+          alignItems: 'center',
+        }}
+      >
+        <MegaphoneIcon
+          sx={{ height: '50px', width: '50px' }}
+          htmlColor={theme.palette.primary.main}
+        />
+        <Typography variant="h6" component="p">
+          {tr.GeneralNotification.noResults}
+        </Typography>
+      </Box>
+    );
+  }
+
+  return (
+    <Box sx={{ overflowY: 'auto' }}>
+      {notifications.map((notification) => (
+        <Box
+          component="article"
+          key={notification.id}
+          sx={notificationListStyle}
+        >
+          <Accordion disableGutters>
+            <AccordionSummary
+              sx={{
+                marginLeft: 0,
+              }}
+              expandIcon={<ExpandMore />}
+            >
+              <Box
+                display="flex"
+                flex={1}
+                justifyContent="space-between"
+                alignItems="center"
+              >
+                <Typography variant="h5" component="h1" color="primary">
+                  {notification.title}
+                </Typography>
+                <Typography color={'#c4c4c4'} fontSize={'14px'}>
+                  {notification.publisher},{' '}
+                  {new Date(notification.createdAt).toLocaleDateString()}
+                </Typography>
+              </Box>
+            </AccordionSummary>
+            <AccordionDetails>
+              <ReactMarkdown>{notification.message}</ReactMarkdown>
+              {editingEnabled && (
+                <Button
+                  size="small"
+                  variant="outlined"
+                  onClick={() =>
+                    editing && notification.id === activeNotification?.id
+                      ? onCancel()
+                      : onEdit(notification)
+                  }
+                  {...(!(
+                    editing && notification.id === activeNotification?.id
```

Comment:
[nit] I think this is a bit of a mouthful, especially with the negation... Also this condition is used in several places, so maybe add some `editingThisNotification` variable to the start of the map function to make it a bit easier to read?